lib/node_modules/@stdlib/math/base/special/riemann-zeta/src/main.c

```diff
@@ -0,0 +1,813 @@
+*
+* ## Notes
+*
+* -   Coefficients should be sorted in ascending degree.
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the rational function
+* @return     evaluated rational function
+*/
+static double rational_p6q6( const double x ) {
+	double ax;
+	double ix;
+	double s1;
+	double s2;
+	if ( x == 0.0 ) {
+		return -10.39489505733089;
+	}
+	if ( x < 0.0 ) {
+		ax = -x;
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
+		s1 = -10.39489505733089 + (x * (-2.858272196711067 + (x * (-0.34772826653924577 + (x * (-0.025115606465534634 + (x * (-0.001194591734169687 + (x * (-0.00003825293235079675 + (x * (-7.855236337967234e-7 + (x * -8.214657090954655e-9)))))))))))));
+		s2 = 1.0 + (x * (0.2081963335726719 + (x * (0.019568765731720502 + (x * (0.0011107963810248593 + (x * (0.000040850774626603926 + (x * (9.555611230656935e-7 + (x * (1.185071534740229e-8 + (x * 2.226094836273526e-15)))))))))))));
+	} else {
+		ix = 1.0 / x;
+		s1 = -8.214657090954655e-9 + (ix * (-7.855236337967234e-7 + (ix * (-0.00003825293235079675 + (ix * (-0.001194591734169687 + (ix * (-0.025115606465534634 + (ix * (-0.34772826653924577 + (ix * (-2.858272196711067 + (ix * -10.39489505733089)))))))))))));
+		s2 = 2.226094836273526e-15 + (ix * (1.185071534740229e-8 + (ix * (9.555611230656935e-7 + (ix * (0.000040850774626603926 + (ix * (0.0011107963810248593 + (ix * (0.019568765731720502 + (ix * (0.2081963335726719 + (ix * 1.0)))))))))))));
+	}
+	return s1 / s2;
+}
+
+// END: rational_p6q6
+
+/* End auto-generated functions. */
+
+/**
+* Evaluates the Riemann zeta function using a polynomial series.
+*
+* ## References
+*
+* -   P. Borwein. "An Efficient Algorithm for the Riemann Zeta Function". Canadian Mathematical Society, Conference Proceedings. See algorithm [3][p155].
+*
+* [p155]: http://www.cecm.sfu.ca/personal/pborwein/PAPERS/P155.pdf
+*
+* @param s    input value
+* @return     function value
+*
+* @example
+* double out = series( 3.0 );
+* // returns ~1.202
+*/
+static double series( const double s ) {
+	uint8_t sign;
+	double term;
+	double sum;
+	double tmp;
+	int32_t N2;
+	int32_t i;
+
+	sum = 0.0;
+	sign = 1;
+	for ( i = 0; i < N; i++ ) {
+		sum += sign * NEG_TWO_N / stdlib_base_pow( (double)( i + 1 ), s );
+		sign *= -1; // flip the sign
+	}
+	tmp = 1.0;
+	term = 1.0;
+	N2 = 2 * N;
+	for ( i = N; i <= N2 - 1; i++ ) {
+		sum += sign * ( tmp - TWO_N ) / stdlib_base_pow( (double)( i + 1 ), s );
+		sign *= -1; // flip the sign
+		term *= N2 - i;
+		term /= i - N + 1.0;
+		tmp += term;
+	}
+	return sum / ( TWO_N * stdlib_base_powm1( 2.0, 1.0 - s ) );
+}
+
+/**
+* Evaluates the Riemann zeta function for an odd positive integer.
+*
+* @param s    input value
+* @return     function value
+*
+* @example
+* double out = zeta( 3.0 );
+* // returns ~1.202
+*/
+static double zeta( const double s ) {
+	int32_t idx;
```

Comment:
Should `idx` be a 32-bit integer? At L590, `s` is double, resulting in the entire expression being implicitly promoted. What does Boost do here?

Reply:
In `boost`, `idx` is `unsigned`.

Reply:
Okay, so at L590, we're performing explicit truncation.